databuilder/databuilder/extractor/neo4j_extractor.py

```diff
@@ -43,7 +47,22 @@ def init(self, conf: ConfigTree) -> None:
         self.conf = conf.with_fallback(Neo4jExtractor.DEFAULT_CONFIG)
         self.graph_url = conf.get_string(Neo4jExtractor.GRAPH_URL_CONFIG_KEY)
         self.cypher_query = conf.get_string(Neo4jExtractor.CYPHER_QUERY_CONFIG_KEY)
-        self.driver = self._get_driver()
+
+        driver = conf.get(Neo4jExtractor.NEO4J_DRIVER, None)
+        if driver and isinstance(driver, Driver):
+            self.driver = driver
+        elif driver and not isinstance(driver, Driver):
+            msg = f'Driver should be of type neo4j.Driver, but an object of type {type(driver)} was given.'
+            LOGGER.error(msg)
+            raise TypeError(msg)
+        else:
+            self.driver = create_neo4j_driver(uri=self.graph_url,
+                                              max_connection_lifetime=conf.get_int(
```

Comment:
i believe because you're not using self.conf, the default fallbacks will not be read